pkg/provider/aws/knowledgebase/iam.go

```diff
@@ -171,6 +175,18 @@ var IamKB = knowledgebase.Build(
 			return nil
 		},
 	},
+	knowledgebase.EdgeBuilder[*resources.IamPolicy, *resources.PrivateDnsNamespace]{
+		Configure: func(policy *resources.IamPolicy, namespace *resources.PrivateDnsNamespace, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			if policy.Policy == nil {
+				policy.Policy = resources.CreateAllowPolicyDocument([]string{"servicediscovery:DiscoverInstances"}, []core.IaCValue{{Property: core.ALL_RESOURCES_IAC_VALUE}})
+				return nil
+			}
+			statement := resources.CreateAllowPolicyDocument([]string{"servicediscovery:DiscoverInstances"}, []core.IaCValue{{Property: core.ALL_RESOURCES_IAC_VALUE}}).Statement
```

Comment:
the dupes dont hurt outside of character count here, but we do have a deduplicate method for statements. Ill make a helper function for this so that we can reduce footprint